cmd/otelcol/main.go

```diff
@@ -164,10 +156,9 @@ func checkRuntimeParams() {
 	// Check if memory ballast flag was passed
 	// If so, ensure memory ballast env var is not set
 	// Then set memory ballast and limit properly
-	ballastSize := getKeyValue(args, "--mem-ballast-size-mib")
+	ballastSize := getKeyValue(os.Args[1:], "--mem-ballast-size-mib")
 	if ballastSize != "" {
-		config = os.Getenv(ballastEnvVarName)
-		if config != "" {
+		if os.Getenv(ballastEnvVarName) != "" {
 			log.Fatalf("Both %v and '--config' were specified, but only one is allowed", ballastEnvVarName)
```

Comment:
```suggestion
			log.Fatalf("Both %v and '--mem-ballast-size-mib' were specified, but only one is allowed", ballastEnvVarName)
```

unrelated but currently incorrect